pkg/tests/BUILD

```diff
@@ -28,6 +28,82 @@ pkg_tar(
     package_dir_file = "testdata/test_tar_package_dir_file.txt"
 )
 
+pkg_zip(
+    name = "test_zip_empty",
+    srcs = [],
+)
+
+# The contents of this file should equal test_zip_empty
+pkg_zip(
+    name = "test_zip_empty_different_extension",
+    srcs = [],
+    extension = "otherkindofzip",
+)
+
+pkg_zip(
+    name = "test_zip_basic",
+    srcs = [
+        "testdata/hello.txt",
+        "testdata/loremipsum.txt",
+    ],
+)
+
+pkg_zip(
+    name = "test_zip_timestamp",
+    srcs = [
+        "testdata/hello.txt",
+    ],
+    timestamp = 1234567890,
+)
+
+# This should be equal to test_zip_basic because timestamps are rounded
+# up to Jan 1 1980.
+pkg_zip(
+    name = "test_zip_basic_timestamp_before_epoch",
+    srcs = [
+        "testdata/hello.txt",
+        "testdata/loremipsum.txt",
+    ],
+    timestamp = 0,
+)
+
+# All of these should produce the same zip file.
+[pkg_zip(
+    name = "test_zip_package_dir" + str(idx),
+    srcs = [
+        "testdata/hello.txt",
+        "testdata/loremipsum.txt",
+    ],
+    package_dir = package_dir,
+) for (idx, package_dir) in enumerate([
+    "abc/def",
+    "/abc/def",
+    "/abc/def/",
+])]
+
+py_test(
+    name = "zip_test",
+    srcs = [
+        "zip_test.py",
+    ],
+    data = [
+        "test_zip_empty.zip",
+        "test_zip_basic.zip",
+        "test_zip_package_dir0.zip",
+        "test_zip_timestamp.zip",
+
+        # these could be replaced with diff_test() rules (from skylib)
```

Comment:
I didn't want to try bringing skylib in inside this PR (especially since it seems like you're up to a lot of stuff around pacakging/federation/bringing this into Bazel core etc.) Let me know what you think.

Reply:
I am not sure diff_test is right either. What you want is a test like this

zip_content_test(
    name = "test_zip_package_dir0_test"
    zip_file = "test_zip_package_dir0.zip"
    expected = [
          "rwxr-xr-x foo",
           "...."
    ]
)

Rather than diffing the content directly, diff the semantic meaning of the content.
But that is worth doing for all the packaging things at once. If we can get a stanard comparison format, the same diff and reporting tool could be used for any of them.

Reply:
Ah, I was thinking `diff_test` because I'm nervous about cross-machine/cross-OS determinism. The zip format has some degrees of freedom where these kind of differences could theoretically sneak in. Checking in a (small) pre-generated zip could guard against that.

`zip_content_test` (etc.) seems cool. I wonder if `pkg_library` etc. could be used for `expected`?

Reply:
> The zip format has some degrees of freedom where these kind of differences could theoretically sneak in.

We want to work that out of the process. Or, at least, to make sure that any bits which are different are no-ops. 
In any case, diff_test does not seem suited for binary comparisons